addDependencyTracker.js

```diff
@@ -0,0 +1,55 @@
+module.exports = function addDependencyTracker(plugin, enableInvalidation) {
+  if (plugin.prototype && plugin.prototype.transform) {
+    // we don't track dependencies for legacy plugins.
+    return plugin;
+  }
+  if (!enableInvalidation) {
+    // Dependency invalidation isn't enabled, so no need to track.
+    return plugin;
+  }
+  if (typeof plugin.getDependencies === "function") {
+    // Don't add the dependency tracker if it's already added
+    return plugin;
+  }
+  // This variable in our closure allows us to share dependencies from
+  // the ast plugin that we can't access with the ast plugin generator that
+  // we can access.
+  let lastDependencies = {};
+  let trackedPlugin = (env) => {
+    let realPlugin = plugin(env);
+    let visitors = realPlugin.visitor;
+    let origProgram = visitors.Program;
+    let origEnter, origExit;
+    if (origProgram) {
+      if (typeof origProgram === "function") {
+        origEnter = origProgram;
+        origExit = undefined;
+      } else {
+        origEnter = origProgram.enter;
+        origExit = origProgram.exit;
+      }
+    }
+    visitors.Program = {
+      enter: (node) => {
+        let fileName = node.loc.source;
```

Comment:
Why are we reading the fileName from `node.loc.source` here? Is it populated correctly?

Any reason not to use `env.meta.moduleName` instead?

Reply:
I wasn't sure that would be unique enough. I don't have a good sense of whether a template compiler instance would ever be shared by more than one instance of this addon in the future. If you're ok with this, I am.

Reply:
>  Is it populated correctly?

I added code in [this commit](https://github.com/ember-cli/ember-cli-htmlbars/pull/238/commits/d73ffdaad9b8df1225364c5b156247174545c3b6) to properly set up the glimmer parser.

Reply:
Ya, my comment in that area is what I mean. Specifically Ember < 3.10 will never have `node.loc.source`.

Reply:
I'm using `env.meta.moduleName` now.